### PR TITLE
Pickle quotes directly as expressions in TASTY

### PR DESCRIPTION
**Pull request by nicolasstucki**
* We avoid having to do owner changes after unpickling
* The pickled code is slightly smaller

Replaces #4242

### CODE REVIEW

**Comment by nicolasstucki** [comment]
test performance with #quotes please

**Comment by nicolasstucki** [comment]
And performance improvements all around :)

**Comment by nicolasstucki** [comment]
And performance improvements all around :)

File: compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala [hunk 1]
```diff
@@ -73,11 +73,18 @@ class TreeUnpickler(reader: TastyReader,
    */
   def enterTopLevel(roots: Set[SymDenotation])(implicit ctx: Context): Unit = {
     this.roots = roots
-    var rdr = new TreeReader(reader).fork
+    val rdr = new TreeReader(reader).fork
     ownerTree = new OwnerTree(NoAddr, 0, rdr.fork, reader.endAddr)
     rdr.indexStats(reader.endAddr)
   }
 
+  def unpickleExpr()(implicit ctx: Context): Tree = {
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala]: Document what this method does, how it differs from unpickle(), when it is used

**Comment by smarter** [comment]
I wonder if we could always unpickle as an expression and get rid of the other way to unpickle.

**Comment by nicolasstucki** [comment]
Do you mean get rid of `enterTopLevel`?

**Comment by nicolasstucki** [comment]
I will merge the two unpicking ways in another PR

**Pull request merged**